### PR TITLE
CB16

### PR DESCRIPTION
**Pull request by cansibi**

### CODE REVIEW

File: src/middleware/createCompanyAccount.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import express, { Request, Response } from 'express'
+import bodyParser from 'body-parser'
+import mongoose from 'mongoose'
+import { User } from '../models/User'
+
+const app = express()
+app.use(bodyParser.json())
+const connectionString = 'mongodb+srv://AGA:lki09bns37ANZO7U@crank-bit.00j0aqc.mongodb.net/'
```
Comment (Qiming-Liu) [inline src/middleware/createCompanyAccount.ts]: we already have these logic in app.ts

File: src/middleware/createCompanyAccount.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      } else {
+        // Create new user
+        const newUser = new User({ username, password })
+        await newUser.save()
+        res.json({ isValide: true, message: 'Success' })
+      }
+    } catch (error) {
+      res.json({ error: 'server error' })
+    }
+  })
```
[the remark this answers is not in the document]
Comment (Qiming-Liu) [reply src/middleware/createCompanyAccount.ts:34]: https://codeql.github.com/codeql-query-help/javascript/js-missing-rate-limiting/

**Review by Qiming-Liu — CHANGES_REQUESTED** [review]

**Comment by Qiming-Liu** [comment]
Mark as draft, because this need refactor, can not merge

File: src/models/User.ts [hunk 3]
```diff
@@ -28,6 +29,12 @@ const UserSchema = new mongoose.Schema({
     minlength: 6,
     select: false,
   },
+  companyUrls: [
```
Comment (LH-Song) [inline src/models/User.ts:32]: may need id here

File: src/controllers/createCompanyUrl.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { Request, Response } from 'express'
+import { StatusCodes } from 'http-status-codes'
+import Joi from 'joi'
+import { User, IUser } from '../models/User'
+
+const createCompanyUrl = async (req: Request, res: Response): Promise<void> => {
```
Comment (LH-Song) [inline src/controllers/createCompanyUrl.ts:6]: I believe we already got a company controllers handle this, check main plz

**Pull request closed without merging**